enacts/wat_bal/maproom_monit.py

```diff
@@ -0,0 +1,513 @@
+
+import psycopg2
+from psycopg2 import sql
+import shapely
+from shapely import wkb
+from shapely.geometry.multipolygon import MultiPolygon
+
+CONFIG_GLOBAL = pingrid.load_config(os.environ["CONFIG"])
+CONFIG = CONFIG_GLOBAL["wat_bal_monit"]
+
+PFX = CONFIG["core_path"]
+TILE_PFX = "/tile"
+
+with psycopg2.connect(**CONFIG_GLOBAL["db"]) as conn:
+    s = sql.Composed([sql.SQL(CONFIG_GLOBAL['shapes_adm'][0]['sql'])])
+    df = pd.read_sql(s, conn)
+    clip_shape = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))[0]
+
+# Reads daily data
+
+DR_PATH = CONFIG["rr_mrg_zarr_path"]
+RR_MRG_ZARR = Path(DR_PATH)
+rr_mrg = calc.read_zarr_data(RR_MRG_ZARR)
+
+# Assumes that grid spacing is regular and cells are square. When we
+# generalize this, don't make those assumptions.
+RESOLUTION = rr_mrg['X'][1].item() - rr_mrg['X'][0].item()
+# The longest possible distance between a point and the center of the
+# grid cell containing that point.
+
+SERVER = flask.Flask(__name__)
+APP = dash.Dash(
+    __name__,
+    server=SERVER,
+    external_stylesheets=[
+        dbc.themes.BOOTSTRAP,
+        "https://use.fontawesome.com/releases/v5.12.1/css/all.css",
+    ],
+    url_base_pathname=f"{PFX}/",
+    meta_tags=[
+        {"name": "description", "content": "Water Balance Maproom"},
+        {"name": "viewport", "content": "width=device-width, initial-scale=1.0"},
+    ],
+)
+APP.title = CONFIG["app_title"]
+
+APP.layout = layout_monit.app_layout()
+
+
+def adm_borders(shapes):
+    with psycopg2.connect(**CONFIG_GLOBAL["db"]) as conn:
+        s = sql.Composed(
+            [
+                sql.SQL("with g as ("),
+                sql.SQL(shapes),
+                sql.SQL(
+                    """
+                    )
+                    select
+                        g.label, g.key, g.the_geom
+                    from g
+                    """
+                ),
+            ]
+        )
+        df = pd.read_sql(s, conn)
+
+    df["the_geom"] = df["the_geom"].apply(lambda x: wkb.loads(x.tobytes()))
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    for i in df.index:
+        shapes[i]['label'] = df['label'][i]
+    return {"features": shapes}
+
+
+def make_adm_overlay(adm_name, adm_sql, adm_color, adm_lev, adm_weight, is_checked=False):
+    border_id = {"type": "borders_adm", "index": adm_lev}
+    return dlf.Overlay(
+        dlf.GeoJSON(
+            id=border_id,
+            data=adm_borders(adm_sql),
+            options={
+                "fill": True,
+                "color": adm_color,
+                "weight": adm_weight,
+                "fillOpacity": 0
+            },
+        ),
+        name=adm_name,
+        checked=is_checked,
+    )
+
+
+@APP.callback(
+    Output("time_selection", "options"),
+    Output("time_selection", "value"),
+    Input("planting_day","value"),
```

Comment:
Where is this input defined? I can't find anything with this id in layout_monit.py.

Reply:
controls.Date created 2 ids one for day and one for month. Search for Date or for planting_